network-api/networkapi/wagtailcustomization/tests/test_localized_redirects.py

```diff
@@ -0,0 +1,84 @@
+from django.test.utils import override_settings
+from wagtail.contrib.redirects.models import Redirect
+
+from networkapi.wagtailpages.factory import buyersguide as buyersguide_factories
+from networkapi.wagtailpages.tests import base as test_base
+
+
+# Safeguard against the fact that static assets and views might be hosted remotely,
+# see https://docs.djangoproject.com/en/3.1/topics/testing/tools/#urlconf-configuration
+@override_settings(STATICFILES_STORAGE="django.contrib.staticfiles.storage.StaticFilesStorage")
+class LocalizedRedirectTests(test_base.WagtailpagesTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.buyersguide_homepage = buyersguide_factories.BuyersGuidePageFactory(
+            parent=cls.homepage,
+        )
+        cls.content_index = buyersguide_factories.BuyersGuideEditorialContentIndexPageFactory(
+            parent=cls.buyersguide_homepage,
+        )
+
+    def test_redirect(self):
+        """Check that we are redirected to the localized version
+        of the homepage when a Redirect object exists.
+        In this example, a Redirect object exists for /test/ -> /
+        so we expect to be redirected to /en/ when we visit /test/
+        """
+        redirect = Redirect(old_path="/test", redirect_link="/final")
+        redirect.save()
+        response = self.client.get("/test/", follow=True)
+
+        self.assertEqual(response.redirect_chain, [("/final", 301), ("/en/final/", 302)])
```

Comment:
What happens in this case when we visit `/en/test/` and what happens when we visit `/fr/test/`?

Reply:
Those will 404, I've added that into this test in 725275b15afaffc29162944df377c43151128b97 as a way of stating this is what we expect